src/validation/coinbase.cairo

```diff
@@ -102,19 +127,87 @@ fn validate_coinbase_witness(witness: Span<ByteArray>) -> Result<(), ByteArray>
     Result::Ok(())
 }
 
+
 /// Return BTC reward in SATS
 fn compute_block_reward(block_height: u32) -> u64 {
     shr(5000000000_u64, (block_height / 210000_u32))
 }
 
+/// Calculate wtxid commitment
+fn calculate_wtxid_commitment(wtxid_root: Digest) -> Digest {
+    /// Big endian
+    let mut witness_value_byte: ByteArray = "";
+    witness_value_byte.append_word(WITNESS_VALUE, 32);
+
+    let wtxid_root_bytes: ByteArray = wtxid_root.into();
+
+    let mut res = ByteArrayTrait::concat(@wtxid_root_bytes.rev(), @witness_value_byte);
+
+    double_sha256_byte_array(@res)
```

Comment:
It has to be single sha256, not double

Reply:
@m-kus, I don't think it's single sha256. from the example given [here](https://learnmeabitcoin.com/technical/transaction/wtxid/#commitment) double sha256 was used and you can check the wtxid commitment on the block to verify it.

Reply:
Ah, right